### PR TITLE
Fix content change timing issue on Shady DOM

### LINKED ISSUE #120: “Danger” theme variant
Offer a `danger` theme variant which is an alias for the existing `error` variant (or the other way around). The semantics of “error” are not good for buttons, and the expected naming would be “danger” (common in many other UI libraries).

Previously, I wanted to keep the variants synonymous with the color properties, which are "primary", "error" and "success". It was actually initially “danger button”, but I thought it might become difficult to know what the “red” variant would be called in different components.

### PR DESCRIPTION
**Pull request by shadikhani**
Fixes vaadin/vaadin-button-flow#120

### CODE REVIEW

**Comment by tomivirkki** [comment]
Also, the fix suggested by @Haprog in https://github.com/vaadin/vaadin-button-flow/issues/120#issuecomment-496909707 works just fine so there's no need to add all these unrelated changes (the invocation of `.assignedNodes()` seems to run the flush and that accidentally fixes the issue).

If this isn't an issue in the web component, it's better to fix this in `vaadin-button-flow` instead (if the flush can be timed properly).

File: src/vaadin-button.html [hunk 1]
```diff
@@ -147,6 +147,7 @@
           this.$.button.setAttribute('role', 'presentation');
 
           this._addActiveListeners();
+          window.ShadyDOM && window.ShadyDOM.flush();
```
Comment (tomivirkki) [inline src/vaadin-button.html]: Missing a test. Something like what's suggested [here](https://github.com/vaadin/vaadin-button-flow/issues/120#issuecomment-496891538) should work as a test case.

If the fix cannot be tested here in the `vaadin-button`, you need to add a proper description as a comment for this addition. Something that explains why this line is needed.

**Comment by tomivirkki** [comment]
Also, the current `Fix #120` comment in the PR description would cause [this totally unrelated issue](https://github.com/vaadin/vaadin-button/issues/120) to get closed.

File: test/vaadin-button_test.html [hunk 2]
```diff
@@ -129,6 +129,11 @@
         vaadinButton.dispatchEvent(new CustomEvent('blur'));
         expect(vaadinButton.hasAttribute('active')).to.be.false;
       });
+
```
Comment (tomivirkki) [inline test/vaadin-button_test.html]: Accidental empty test case?

Comment (shadikhani) [reply test/vaadin-button_test.html]: Yes. removed.

**Pull request merged**